### DISCUSSION TITLE
Diferença do Body no Get e Post

### DISCUSSION THREAD

**Discussion by pabloanizio** [viniciussanchez/RESTRequest4Delphi#177] [Q&A] [answered]
Boa tarde Vinicius.

Você sabe me informa porque a diferença das informações do Body no Get e no Post?

OBS.: 
O Get é normal - No Body chega o json normalmente 
O Post eu estou enviando um arquivo - O body chega com uma Key aleatória

**Comment by pabloanizio** [comment]
Outra coisa... estou usando o node com o multer pra pegar o arquivo

**Comment by viniciussanchez** [comment]
Tem muitas variáveis ai na sua pergunta... difícil saber o que pode ser assim... se puder postar códigos, exemplos etc facilita

**Comment by felipegarbelotte** [comment]
Na verdade eu to tentando utilziar o Request.DataSetAdapter e não estou conseguindo. Ele dá um erro:

[dcc32 Error] DataModule.Usuario.pas(58): E2003 Undeclared identifier: 'DataSetAdapter'

Eu quero alimentar um objeto TFDMemTable e na explicação do curso eu defino o adaptador com o DataSetAdapter, mas o delphi não acha esse cara. Como devo fazer? O código está assim:

======================================================
procedure TDMUsuario.Login(email, senha: string);

var
    resp: IResponse;
    json: TJSONObject;

begin
  try
    json := TJSONObject.Create;
    json.AddPair('email_user', email);
    json.AddPair('senha_user', senha);

    resp := TRequest.New.BaseURL(BASE_URL)
            .Resource('users/login')
            .DataSetAdapter(TabUsuario) <---Erro 2003: Undeclared identifier: 'DataSetAdapter'
            .BasicAuthentication(USER_NAME, PASSWORD)
            .AddBody(json.ToJSON)
            .ContentType('application/json')
            .BasicAuthentication(USER_NAME, PASSWORD)
            .Post;


    if (resp.StatusCode <> 200) then
        raise Exception.Create(resp.Content);
  finally
    json.DisposeOf;
  end;

end;

======================================================


Atenciosamente,
Felipe Garbelotte, Desenvolvedor Nocode.
E-mail: ***@***.******@***.***> | ***@***.******@***.***>
WhatsApp .: 015-99697-5117<https://wa.me/5515996975117>

------Evite o desmatamento, não imprima e-mails-----
________________________________
De: Vinicius Sanchez ***@***.***>
Enviado: quarta-feira, 25 de outubro de 2023 09:28
Para: viniciussanchez/RESTRequest4Delphi ***@***.***>
Cc: Subscribed ***@***.***>
Assunto: Re: [viniciussanchez/RESTRequest4Delphi] Diferença do Body no Get e Post (Discussion #177)


Tem muitas variáveis ai na sua pergunta... difícil saber o que pode ser assim... se puder postar códigos, exemplos etc facilita

**Reply by viniciussanchez** [reply, marked as answer]
Entendi. Ouve uma atualização do RESTRequest4Delphi que mudou esse comportamento. Antigamente o RR4D era preso ao DataSet-Serialize, sendo que nem todo mundo usava o mesmo, ou usava outro componente para fazer esse mecanismo. Por isso tiramos essa dependência, e criamos que o chamamos de Adapters. Se olhar a documentação aqui no github vai ver melhor sobre isso.

![image](https://github.com/viniciussanchez/RESTRequest4Delphi/assets/16382981/a4bec339-0328-4d1f-ac70-6d818870daa2)

Basicamente estamos falando desse cara aqui: https://github.com/viniciussanchez/dataset-serialize-adapter-restrequest4delphi e o uso agora ficaria assim:

![image](https://github.com/viniciussanchez/RESTRequest4Delphi/assets/16382981/f5ab185d-6bf0-4f72-bcd9-7e2ec3be6d3c)

**Comment by felipegarbelotte** [comment]
Boa tarde Vinicius.

Muito obrigado, deu certinho!


Qualquer dúvida entre em contato comigo.
***@***.***>
Atenciosamente,
Felipe Garbelotte, Desenvolvedor Nocode.
E-mail: ***@***.******@***.***> | ***@***.******@***.***>
WhatsApp .: 015-99697-5117<https://wa.me/5515996975117>

------Evite o desmatamento, não imprima e-mails-----
________________________________
De: Vinicius Sanchez ***@***.***>
Enviado: sexta-feira, 27 de outubro de 2023 15:54
Para: viniciussanchez/RESTRequest4Delphi ***@***.***>
Cc: felipegarbelotte ***@***.***>; Comment ***@***.***>
Assunto: Re: [viniciussanchez/RESTRequest4Delphi] Diferença do Body no Get e Post (Discussion #177)


Entendi. Ouve uma atualização do RESTRequest4Delphi que mudou esse comportamento. Antigamente o RR4D era preso ao DataSet-Serialize, sendo que nem todo mundo usava o mesmo, ou usava outro componente para fazer esse mecanismo. Por isso tiramos essa dependência, e criamos que o chamamos de Adapters. Se olhar a documentação aqui no github vai ver melhor sobre isso.

[image]<https://user-images.githubusercontent.com/16382981/278734142-a4bec339-0328-4d1f-ac70-6d818870daa2.png>

Basicamente estamos falando desse cara aqui: https://github.com/viniciussanchez/dataset-serialize-adapter-restrequest4delphi e o uso agora ficaria assim:

[image]<https://user-images.githubusercontent.com/16382981/278734044-f5ab185d-6bf0-4f72-bcd9-7e2ec3be6d3c.png>
